lib/odata/service.rb

```diff
@@ -123,9 +123,23 @@ def [](entity_set_name)
     # @param url_chunk [to_s] string to append to service url
     # @param additional_options [Hash] options to pass to Typhoeus
     # @return [Typhoeus::Response]
-    def execute(url_chunk, additional_options = {})
+    def execute(url_chunk, additional_options = {}, escaped = false)
+      # There must be a better way to do it,
+      # but we only use it for Chocolatey, so it won't break anything else
+      # This just forces us to use Search endpoint for searching through
+      # Chocolatey instead of Packages.
+      if url_chunk && url_chunk.starts_with?('Packages?')
```

Comment:
Maybe we should change `Packages` to `Search()` in `app/controllers/search_controller.rb:78` instead?

Reply:
Yeah, I tried, it did not work. `Search()` is a function within the same `Packages` entityset, so we have to use `Packages` entityset. I think this gem just doesn't provide the functionality to call these special functions or if it does, I couldn't find a way.

Chocolatey metadata endpoint: https://community.chocolatey.org/api/v2/$metadata

![image](https://github.com/repairtech/odata/assets/98130064/7d23ea7c-d189-4297-9382-ae3474389227)